### ISSUE TITLE
Distorted image when apply at least on filter

### ISSUE THREAD

**Issue by lillogoal** [burhanrashid52/PhotoEditor#291]
Hi,

i have the problem in subject. My Flow is:
Download with glide a bitmap and after i will send to PhotoEditor library. When i select at least one filter (gray,rotate, ecc...) the image is distorted.

Before

![Screenshot_1605772807](https://user-images.githubusercontent.com/4365815/99637939-d59edb80-2a45-11eb-9b69-48c5ab0496d0.png)

After

![Screenshot_1605772871](https://user-images.githubusercontent.com/4365815/99637963-db94bc80-2a45-11eb-82b1-4024a43713c8.png)

**Comment by rohit6027** [comment]
You can fix this issue by saving the captured image in the gallery and use that file path because if you take an image directly from the camera then the size of that image is very small and this is the reason the image is getting destored.

**Comment by lillogoal** [comment]
@rohit6027 i'm downloading image with glide. 

     ```
final GlideUrl glideUrl = GenericUtility.getGlideUrlFromData(BackendMethods.URL_IMG_TIMELINE_PATH+ picMatch.get_key(), MyApplication.getUser().getWtkn());
                Glide.with(EditImageActivity.this)
                        .load(glideUrl)
                        .asBitmap()
                        .into(new SimpleTarget<Bitmap>() {
                            @Override
                            public void onResourceReady(Bitmap resource, GlideAnimation<? super Bitmap> glideAnimation) {
                                mPhotoEditorView.getSource().setImageBitmap(resource);
                            }});
```

so i'm using bitmap. do you think that correct way is going to save in sd and after ?

**Comment by rohit6027** [comment]
It doesn't matter how you are loading the image in image view the only thing matter is what file path are you using.
how are you getting belwo path in onActvityResult. 

 GenericUtility.getGlideUrlFromData(BackendMethods.URL_IMG_TIMELINE_PATH+ picMatch.get_key(),

**Comment by lillogoal** [comment]
@rohit6027 mmm path is my server next i download the image and setBitmap I don't understand where the problem is

**Comment by lillogoal** [comment]
@rohit6027 any updates ? ty

**Comment by rohit6027** [comment]
@lillogoal your case is different but the problem I was facing is the same. I solved my problem after saving the actual image size in gallery when the user clicks an image from the camera.

**Comment by lillogoal** [comment]
> @lillogoal your case is different but the problem I was facing is the same. I solved my problem after saving the actual image size in gallery when the user clicks an image from the camera.

so you suggest me to save in external directory and after loads in photoeditor view ? can you post an example of saving and loading image ? thanks

**Comment by rohit6027** [comment]
@lillogoal  no need to save image in an external directory. It will be great if you share a complete image link that you are trying loading in the photo editor view.

**Comment by lillogoal** [comment]
> @lillogoal no need to save image in an external directory. It will be great if you share a complete image link that you are trying loading in the photo editor view.

sorry but link required auth request with bearer.

i understud image has to save  in gallery right ?

**Comment by rohit6027** [comment]
I am not sure it will work or not but you can try it.

**Comment by lillogoal** [comment]
@rohit6027 yes i can try but you can send me an example of saving in gallery and load from gallery. ty

**Comment by rohit6027** [comment]
In the beginning, I said I am taking an image from the camera but you are taking it from the server.

**Comment by lillogoal** [comment]
yes yes but after taking from server i convert in bitmap so maybe your implementation is ok. I am desperate I have tried everything

**Comment by rohit6027** [comment]
is it happening with every image URL which you getting from the backend ?? if yes then I will create a sample for the same issue else you can share your image URL if possible.

**Comment by rohit6027** [comment]
@lillogoal  its working file with the URL also I am using this url -> [](https://www.webconfs.com/comics/dynamic-vs-static-urls-comic.jpg)

**Comment by lillogoal** [comment]
My flow is: user capture image -> upload on server -> other user download image and need to modify it.

If user in the first step decide to load an image from gallery (low resolution maybe) it works. I tested with your url: "https://www.webconfs.com/comics/dynamic-vs-static-urls-comic.jpg" --> it works.

So in my opinion problem is when user capture and upload image with high resolution.

![Screenshot_1609770302](https://user-images.githubusercontent.com/4365815/103544943-24db8700-4ea1-11eb-8b05-064837184ac0.png)

**Comment by rohit6027** [comment]
yes you are right. how you are capturing the image ?? are you saving actual image size in local storage ??

**Comment by lillogoal** [comment]
```
  PickSetup pickSetup = new PickSetup().setWidth(600).setHeight(400).setGalleryButtonText(getString(R.string.galleria)).setTitle(getString(R.string.choose)).setCancelText(getString(R.string.annulla));
        PickImageDialog.build(pickSetup).show(this).setOnPickResult(new IPickResult() {
            @Override
            public void onPickResult(PickResult pickResult) {
                if (pickResult.getError() == null) {
                    Log.e("PATH",pickResult.getPath());
                    File file = new File(Environment.getExternalStorageDirectory()+File.separator+"img_"+System.currentTimeMillis()+".jpg");

                    UCrop.of(pickResult.getUri(), Uri.fromFile(file))
                            .start(PhotoToBetActivity.this,REQUEST_IMAGE_CROP);
                } else {

                    //Handle possible errors
                    //TODO: do what you have to do with r.getError();
                    Toast.makeText(PhotoToBetActivity.this, pickResult.getError().getMessage(), Toast.LENGTH_LONG).show();
                }
            }
        });
```

I think yes, this is my code, i used a picksetup library, and after selection or capture image user can crop using UCrop library

**Comment by rohit6027** [comment]
remove .setWidth(600).setHeight(400) and try again may be this is the problem.

**Comment by lillogoal** [comment]
You are right!! I'm testing with more and more cases. After tests i will reply with result.

**Comment by Somil-Jain** [comment]
Hi,

I am also facing a similar issue when apply atleast a filter. The image gets stretched from both X and Y axis and seems to be distorted as well. Has anyone found a solution for this?

I have tried to get the child views to check whether we can change the Scale of Image but unable to achieve this.

Can any one help here please.
